code/game/machinery/atmoalter/canister.dm

```diff
@@ -369,115 +320,103 @@ update_flag
-		else
-			if(holding)
-				logmsg = "Valve was <b>opened</b> by [key_name(usr)], starting the transfer into the [holding]<br>"
-			else
-				logmsg = "Valve was <b>opened</b> by [key_name(usr)], starting the transfer into the <font color='red'><b>air</b></font><br>"
-				if(air_contents.toxins > 0)
-					message_admins("[key_name_admin(usr)] opened a canister that contains plasma in [get_area(src)]! (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)")
-					log_admin("[key_name(usr)] opened a canister that contains plasma at [get_area(src)]: [x], [y], [z]")
-				if(air_contents.sleeping_agent > 0)
-					message_admins("[key_name_admin(usr)] opened a canister that contains N2O in [get_area(src)]! (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)")
-					log_admin("[key_name(usr)] opened a canister that contains N2O at [get_area(src)]: [x], [y], [z]")
-		investigate_log(logmsg, "atmos")
-		release_log += logmsg
-		valve_open = !valve_open
-
-	if(href_list["remove_tank"])
-		if(holding)
-			if(valve_open)
-				valve_open = 0
-				release_log += "Valve was <b>closed</b> by [key_name(usr)], stopping the transfer into the [holding]<br>"
-			holding.loc = loc
-			holding = null
-
-	if(href_list["pressure_adj"])
-		var/diff = text2num(href_list["pressure_adj"])
-		if(diff > 0)
-			release_pressure = min(10*ONE_ATMOSPHERE, release_pressure+diff)
-		else
-			release_pressure = max(ONE_ATMOSPHERE/10, release_pressure+diff)
-
-	if(href_list["rename"])
-		if(can_label)
-			var/T = sanitize(copytext(input("Choose canister label", "Name", name) as text|null,1,MAX_NAME_LEN))
-			if(can_label) //Exploit prevention
-				if(T)
-					name = T
+		return
+	init_data_vars()
+	var/can_min_release_pressure = round(ONE_ATMOSPHERE / 10)
+	var/can_max_release_pressure = round(ONE_ATMOSPHERE * 10)
+	. = TRUE
+	switch(action)
+		if("relabel")
+			if(can_label)
+				var/T = sanitize(copytext(input("Choose canister label", "Name", name) as text|null, 1, MAX_NAME_LEN))
+				if(can_label) //Exploit prevention
+					if(T)
+						name = T
+					else
+						name = "canister"
 				else
-					name = "canister"
+					to_chat(usr, "<span class='warning'>As you attempted to rename it the pressure rose!</span>")
+					. = FALSE
+		if("pressure")
+			var/pressure = params["pressure"]
+			if(pressure == "reset")
+				pressure = ONE_ATMOSPHERE
+			else if(pressure == "min")
+				pressure = can_min_release_pressure
+			else if(pressure == "max")
+				pressure = can_max_release_pressure
+			else if(pressure == "input")
+				pressure = input("New release pressure ([can_min_release_pressure]-[can_max_release_pressure] kPa):", name, release_pressure) as num|null
+				if(isnull(pressure))
+					. = FALSE
+			else if(text2num(pressure) != null)
+				pressure = text2num(pressure)
+			if(.)
+				release_pressure = clamp(round(pressure), can_min_release_pressure, can_max_release_pressure)
```

Comment:
This will clamp everything above it, hopefully.

Reply:
That's the idea.